proto/config/config.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+package linkerd2.config;
+
+option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
+
+// ConfigStore provides an abstraction over the underlying configuration storge.
+service ConfigStore {
+  // Return the JSON-serialized global config.
+  rpc GetGlobalConfig(GetParam) returns (GlobalConfig) {}
+
+  // Return the JSON-serialized proxy config.
+  rpc GetProxyConfig(GetParam) returns (ProxyConfig) {}
+
+  // Set the global config. It overwrites the existing value.
+  rpc SetGlobalConfig(GlobalConfig) returns (Result) {}
+
+  // Set the proxy config. It overwrites the existing value.
+  rpc SetProxyConfig(ProxyConfig) returns (Result) {}
+}
+
+message GlobalConfig {
+  string api_addr = 1;
+  string linkerd_namespace = 2;
+  string linkerd_version = 3;
+  string cluster_domain = 4;
+  bool no_init_container = 5;
+  bool ha = 6;
+  bool auto_inject = 7;
+}
```

Comment:
Any particular reason why not all the control plane configs are listed here?

Reply:
I'm still not 100% sure if all control plane configs are _global_. I guess as starter, we can include all the configs, and then add and trim as we go.

Reply:
(I think our last comments overlapped by less than a minute). I agree with starting by keeping the config to the minimal and increment as needed, simply because of the unknown effects of changing some of them during runtime (if we decided to watch them for reasons). 

Regarding `auto_inject` and others like `controller_log_level`, `disable_h2_upgrade`, `ha` etc., which aren't needed for install/inject, I was (jumping the :gun: again) hoping for a mechanism to record all the initial installation configurations for, who knows, if they would be helpful (?) for future debugging. But let's keep the struct small for now.